tests/Yaapii.Atoms.Tests/Map/SortedTest.cs

```diff
@@ -97,46 +99,71 @@ public void DefaultComparerSeemsSane(int index, int expectedKey)
+                     ),
+                    StringComparer.CurrentCultureIgnoreCase
+                ).Values
+            );
+
+            Assert.Equal(
+                ManyOf.New(2, 1),
+                new Sorted<int>(
+                    MapOf.New(
+                        "b", 1,
+                        "a", 2
+                     ),
+                    StringComparer.CurrentCultureIgnoreCase
+                ).Values
+            );
+
+            Assert.Equal(
+                ManyOf.New("2", "1"),
+                new Sorted(
+                    MapOf.New(
+                        "b", "1",
+                        "a", "2"
+                     ),
+                    StringComparer.CurrentCultureIgnoreCase
+                ).Values
             );
-            var sorted = new Sorted<int, int>(unsorted);
-            var keys = new int[3];
-            sorted.Keys.CopyTo(keys, 0);
-            Assert.Equal(expectedKey, keys[index]);
         }
 
         [Fact]
-        public void DeliversSingleValueWhenLazy()
+        public void ListsValuesFromGenericValue()
         {
-            var unsorted = new LazyDict<int, int>(false,
-                new KvpOf<int, int>(1, () => 4),
-                new KvpOf<int, int>(6, () => { throw new Exception("i shall not be called"); }),
-                new KvpOf<int, int>(-5, () => { throw new Exception("i shall not be called"); })
+            Assert.Equal(
+                ManyOf.New(2, 1),
+                new Sorted<int>(
+                    MapOf.New(
+                        "b", 1,
+                        "a", 2
+                     ),
+                    StringComparer.CurrentCultureIgnoreCase
+                ).Values
             );
-            var sorted = new Sorted<int, int>(unsorted);
-            Assert.Equal(4, sorted[1]);
         }
 
         [Fact]
-        public void RejectsBuildingAllValuesByDefault()
+        public void ListsValuesFromStringPairs()
         {
-            var unsorted = new LazyDict<int, int>(false,
-                new KvpOf<int, int>(1, () => 4),
-                new KvpOf<int, int>(6, () => { throw new Exception("i shall not be called"); }),
-                new KvpOf<int, int>(-5, () => { throw new Exception("i shall not be called"); })
+            Assert.Equal(
+                ManyOf.New("2", "1"),
+                new Sorted(
+                    MapOf.New(
+                        "b", "1",
+                        "a", "2"
+                     ),
+                    StringComparer.CurrentCultureIgnoreCase
+                ).Values
             );
-            var sorted = new Sorted<int, int>(unsorted);
-            var ex = Assert.Throws<InvalidOperationException>(() => sorted.Values.GetEnumerator());
-            Assert.Equal("Cannot get all values because this is a lazy dictionary. Getting the values would build all keys. If you need this behaviour, set the ctor param 'rejectBuildingAllValues' to false.", ex.Message);
```

Comment:
I would like to see a test here, to show that the values are not executed, when only sorting by keys. something like this:
```csharp
[Fact]
public void SortsWithoutExecutingValues()
{
   var count = 0;
   new Sorted(
      new MapOf(
         new KvpOf("2", ()=>{ count++; return "b";}),
         new KvpOf("1", ()=>{ count++; return "a";})
      )
   ).Keys;
   Assert.Equal(
      0,
      count
   );
}